### ISSUE TITLE
azure-storage-node/browser/ still has dependency on node.js for createBlockBlobFromStream??

### ISSUE THREAD

**Issue by zijianhuang** [Azure/azure-storage-node#520]
### Which service(blob, file, queue, table) does this issue concern?

Azure Blob Storage

### Which version of the SDK was used?

 azurestoragejs-2.10.101

### What's the Node.js/Browser version?

Chrome Version 69.0.3497.100 (Official Build) (64-bit), and other browsers

### What problem was encountered?

When calling createBlockBlobFromStream() with respective parameters, I got error:

`t.pause is not a function`

### Steps to reproduce the issue?

I had made a ReadableStream from a base64 DataURI, and pass the stream to createBlockBlobFromStream, however, apparently the lib expects the stream is stream.ReadableStream defined in NodeJS, rather than ReadableStream defined in https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream. Since the error is apparently thrown at this place:

```js
BlobService.prototype._uploadBlobFromStream = function (isNewBlob, container, blob, blobType, stream, streamLength, optionsOrCallback, callback) {
  var options;
  azureutil.normalizeArgs(optionsOrCallback, callback, function (o, c) { options = o; callback = c; });
  options.speedSummary = options.speedSummary || new SpeedSummary(blob);

  if (blobType === BlobConstants.BlobTypes.BLOCK) {
    // default to true, unless explicitly set to false
    options.storeBlobContentMD5 = options.storeBlobContentMD5 === false ? false : true;
  }

  stream.pause();
```

### Have you found a mitigation/solution?

Using the lib I have already uploaded text and local browser file directly from a Web browser to to Azure Storage without passing data to my ASP.NET Web API backend. However, this createBlockBlobFromStream is not working. 

The data is created in the browser with a web-base whiteboard creating a PNG image in memory. Now I may have to upload the PNG to my Web API backend, and the backend then uploads the data to Azure Storage. Rather wasteful, not nice.

**Comment by jiacfan** [comment]
Hi, @zijianhuang 

Thanks for reaching us, and sorry that this is not supported currently.

Browser `ReadStream` is an experimental standard(https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream), which is described nascent and immature.

We'll add this feature to candidate planning items, and thanks for the suggestions!

Best Regards,
Jiachen

**Comment by zijianhuang** [comment]
I think that the lib should at least support Blob and ArrayBuffer which are mature, as "@azure/storage-blob": "^10.0.0-preview" does, which was released around 10 days ago.

In my Angular 2+ app, I had actually created a data access layer in my app, which wrap the lib into a service with functions returning Observable<T>. This has been running well for a year, until recently I need to upload blob.

The data access layer was transforming the callbacks into Observable<T>, thus was similar to what done @azure/storage-blob" which is using Promise<T>.

So I am not really asking supporting ReadableStream which is immature, but supporting Blob and ArrayBuffer.
